plugins/groups_manager.py

```diff
@@ -0,0 +1,307 @@
+        section.add_argument('--mandatory', action='store_true',
+                             help=_('add packages to the mandatory section'))
+        section.add_argument('--optional', action='store_true',
+                             help=_('add packages to the optional section'))
+        section.add_argument('--remove', action='store_true', default=False,
+                             help=_('remove packages from group instead of adding them'))
+        parser.add_argument('--dependencies', action='store_true',
+                            help=_('include also direct dependencies for packages'))
+
+        parser.add_argument("packages", nargs='*', metavar='PACKAGE',
+                            help=_('package specification'))
+
+    def configure(self):
+        demands = self.cli.demands
+
+        if self.opts.packages:
+            demands.sack_activation = True
+            demands.available_repos = True
+            demands.load_system_repo = False
+
+        # handle --merge option (shortcut to --load and --save the same file)
+        if self.opts.merge:
+            self.opts.load.insert(0, self.opts.merge)
+            self.opts.save.append(self.opts.merge)
+
+        # check that group is specified when editing is attempted
+        if (self.opts.description
+                or self.opts.display_order
+                or self.opts.translated_name
+                or self.opts.translated_description
+                or self.opts.user_visible is not None
+                or self.opts.packages):
+            if not self.opts.id and not self.opts.name:
+                raise dnf.cli.CliError(
+                    _("Can't edit group without specifying it (use --id or --name)"))
+
+    def load_input_files(self):
+        """
+        Loads all input xml files.
+        Returns True if at least one file was successfuly loaded
+        """
+        for file_name in self.opts.load:
+            file_comps = libcomps.Comps()
+            try:
+                if file_name.endswith('.gz'):
+                    # libcomps does not support gzipped files - decompress to temporary
+                    # location
+                    with gzip.open(file_name) as gz_file:
+                        with tempfile.NamedTemporaryFile() as temp_file:
+                            shutil.copyfileobj(gz_file, temp_file)
+                            # The xml written to temp_file  might remain in buffers.
+                            # Force writing it to disk.
+                            temp_file.file.flush()
+                            os.fsync(temp_file.file.fileno())
+                            file_comps.fromxml_f(temp_file.name)
+                else:
+                    file_comps.fromxml_f(file_name)
+            except (IOError, OSError, libcomps.ParserError) as err:
+                # gzip module raises OSError on reading from malformed gz file
+                logger.error(_("Can't load file \"{}\": {}").format(file_name, err))
```

Comment:
How about putting this line into the description of `dnf.exceptions.Error()` instead of logging it here? The error is without a description and here it is logged, I know the order will be different but it seems cleaner to me.

Reply:
Yes, thanks - I've changed it according to your suggestion.